### PR TITLE
Standardize detection of traceId for 'corr' log field

### PR DESCRIPTION
**Pull request by irinaepshteyn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,10 +18,6 @@ Utility  for request tracing, log enrichment and auditing.
 
 # HTTP Request tracing
 
-## Populate zipkin HTTP headers for [tracing](opentracing.io)
-
-This filter initializes an HTTP header(X-B3-TraceID) for tracing, if not already present. The header is also added in the returned response.  Note that if you are already using another library for propogating headers, this will have no effect.
```
Comment (akarollil) [inline README.md:23]: What if there are library users who now expect the X-B3-TraceID header to be added by the library? They will not have the header any more in responses?

File: src/main/java/com/ge/predix/log/filter/LogFilter.java [hunk 2]
```diff
@@ -185,16 +180,6 @@ private String setZoneId(final HttpServletRequest request) {
         return zoneId;
     }
 
-    private String setCorrelationId(final HttpServletRequest request, final HttpServletResponse response) {
-        String correlationId = request.getHeader(CORRELATION_HEADER_NAME);
-        if (StringUtils.isEmpty(correlationId)) {
-            correlationId = UUID.randomUUID().toString();
-        }
-        MDC.put(CORRELATION_HEADER_NAME, correlationId);
-        response.setHeader(CORRELATION_HEADER_NAME, correlationId);
```
Comment (akarollil) [inline src/main/java/com/ge/predix/log/filter/LogFilter.java:194]: Maybe we can still keep setting the X-B3-TraceID header to maintain backwards compatibility for those folk who might now depend on the header in responses?

File: src/test/java/com/ge/predix/log4j1/PredixLayoutTest.java [hunk 3]
```diff
@@ -276,13 +279,124 @@ public void testPredixEncoderWithNonStringMessageAndMessageLineSeparator() {
...
+                          + ZONE_VALUE + "\",\"corr\":\"" + CORRELATION_VALUE_OTHER + "\",\"appn\":\"" + APP_NAME_VALUE
+                          + "\",\"dpmt\":\"" + APP_ID_VALUE + "\",\"inst\":\"" + INSTANCE_ID_VALUE + "\",\"tid\":\"" + THREAD_NAME
+                          + "\",\"mod\":\"" + FILE_NAME + "\",\"lvl\":\"" + Level.INFO.toString()
+                          + "\",\"msg\":{\"width\":4,\"length\":3,\"units\":\"inches\",\"height\":5}}\n";
+
+        PredixLayout predixLayout = new PredixLayout();
+        predixLayout.setMessageLineSeparatorRegex(null);
+        predixLayout.setCorrelationKey(CORRELATION_KEY_OTHER);
+        String actual = predixLayout.format(input);
+        System.out.println(actual);
```
Comment (akarollil) [inline src/test/java/com/ge/predix/log4j1/PredixLayoutTest.java:301]: NIT: Guess we can remove this line

**Review by akarollil — COMMENTED** [review]
I have some comments about backwards compatibility for library consumers. I will let @metacubed approve.

File: README.md (hunk 1 above)
Comment (irinaepshteyn) [reply README.md:23]: Idea is to make LogFilter rely on other libraries/frameworks to propagate context such as Spring Cloud Sleuth or Splunk Observability.  Besides, I am not aware of any consumers of this library besides our security services.

File: src/test/java/com/ge/predix/log4j1/PredixLayoutTest.java (hunk 3 above)
Comment (irinaepshteyn) [reply src/test/java/com/ge/predix/log4j1/PredixLayoutTest.java:301]: It's harmless and makes it easier to debug failed tests.

File: README.md (hunk 1 above)
Comment (metacubed) [reply README.md:23]: @akarollil, the response header is now added by the tracing library. That's the right place for that functionality. It was only added here as a temporary workaround after Spring Sleuth removed its response header functionality.

File: src/main/java/com/ge/predix/log/filter/LogFilter.java (hunk 2 above)
Comment (metacubed) [reply src/main/java/com/ge/predix/log/filter/LogFilter.java:194]: @akarollil see comment above

**Review by metacubed — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (akarollil) [reply README.md:23]: I understand the change and I like the header injection removal, but just thinking about users of the library who may be depending on the injection behavior and will now miss the header.

**Pull request merged**